src/UIAutomation/UiaOperationAbstraction/UiaOperationAbstraction.h

```diff
@@ -123,6 +123,55 @@ namespace UiaOperationAbstraction
             }
         }
 
+        template<class TryBody, class CatchBody>
+        void Try(TryBody&& tryBody, CatchBody&& catchBody) const

```

Comment:
I'm curious what your thoughts are on this.

My thinking is instead of adding a `GetCurrentFailureCode` API, we just pass the failure code into the catch block. This serves a couple purposes:

- More similar to C++ where the exception is passed into the catch block.
- Doesn't risk a user calling `GetCurrentFailureCode` from outside a catch block, which I think would cause a fail fast.

I'm thinking pass in a new type, maybe `UiaException`, to the catch block. This new type would have a method `GetFailureCode`, which internally calls either `m_remoteOperation.GetCurrentFailureCode` or `wil::ResultFromCaughtException`. It could potentially have other methods in the future.

What are your thoughts on this?